clearnode/custody.go

```diff
@@ -525,33 +531,44 @@ func (c *Custody) handleClosed(logger Logger, ev *nitrolite.CustodyClosed) {
 		channelAccountID := NewAccountID(channelID)
 		walletAccountID := NewAccountID(channel.Wallet)
 
-		// Transfer from unified account into channel account and then withdraw immediately.
 		ledger := GetWalletLedger(tx, walletAddress)
-		channelAccountBalance, err := ledger.Balance(channelAccountID, asset.Symbol)
+		channelEscrowAccountBalance, err := ledger.Balance(channelAccountID, asset.Symbol)
 		if err != nil {
 			return fmt.Errorf("error fetching channel balance: %w", err)
 		}
 
-		// Withdraw from unified balance if channel was not in joining state.
-		if channelAccountBalance.IsZero() { // If channel balance is not zero, it means the channel was in joining state.
-			if err := ledger.Record(walletAccountID, asset.Symbol, amount.Neg(), nil); err != nil {
-				return fmt.Errorf("error recording balance update for participant: %w", err)
+		// 1. Unlock funds from channel Escrow if not empty
+		if !channelEscrowAccountBalance.IsZero() {
+			if err := ledger.Record(channelAccountID, asset.Symbol, channelEscrowAccountBalance.Neg(), nil); err != nil {
+				return err
+
 			}
-			if err := ledger.Record(channelAccountID, asset.Symbol, amount, nil); err != nil {
-				return fmt.Errorf("error recording balance update for wallet: %w", err)
+			if err := ledger.Record(walletAccountID, asset.Symbol, channelEscrowAccountBalance, nil); err != nil {
+				return err
 			}
-			_, err = RecordLedgerTransaction(tx, TransactionTypeWithdrawal, walletAccountID, channelAccountID, asset.Symbol, amount)
+			_, err := RecordLedgerTransaction(tx, TransactionTypeEscrowUnlock, channelAccountID, walletAccountID, asset.Symbol, channelEscrowAccountBalance)
 			if err != nil {
-				return fmt.Errorf("failed to record transaction: %w", err)
+				return err
 			}
 		}
```

Comment:
I see a pattern here, would be cool to extract such common actions into some function soon
1) record ledger: from
2) record ledger: to
3) record ledger tx: from -> to